node-sass -> dart-sass

illusionfield opened the pull request:
Closing this PR in favor of the consolidated version here:  
➡️ **#329 – Finalize Dart Sass migration & drop direct dependency**

That PR targets `release/5.0.0` and includes all changes from this one, along with updated docs and migration instructions.  
Looking forward to your feedback there!

Hi,

I've spent quite some time working on the integration of Meteor JS and Dart Sass. Given how things are evolving with Dart Sass, I wanted to share my approach. I'm really curious to hear the feedback from the maintainer community on whether this direction is viable.

- It requires Meteor 3
- Dropped the `seba:minifiers-autoprefixer` dependency and aligned it with Meteor’s standard-minifier, which now generates perfect mappings.
- Reworked the `scss-config.json` for improved handling.
- The `includePaths` config option seems less relevant now, but this might need further review to cover any cases I might have missed.
- Due to unique import strategies in both Meteor and Dart Sass, I had to sync up the import functionality, so it’s not fully backward-compatible.  Dart Sass's importer returns a URL object, while its input is an URL string, which would have broken the `@import "{my-package:pretty-buttons}/...";` implementation.

illusionfield committed:
node-sass -> dart-sass

illusionfield committed:
Meteor 2.8 compatibility

illusionfield committed:
Compatibility update

illusionfield committed:
compatibility fix

StorytellerCZ commented on README.md:
```diff
@@ -25,132 +21,112 @@ Package.onUse(function (api) {
-`standard-minifiers` package merges them into one file afterwards.)
+Each compiled source file generates a separate CSS file, which is then merged into one by the `standard-minifiers` package.
 
 ### Importing
```
Looking at the changes here, this requires a migration guide to be included here.

illusionfield replied:
Would something like this be a good starting point? 

[migration-guide.md](https://github.com/user-attachments/files/18708547/migration-guide.md)

dr-dimitru replied:
@illusionfield regarding migration guide:
- Add short description of difference between `@use` and `@forward` with reference to the docs
- Run mdlinter over the file (Bold fonts in headers are rudimentary)
- The rest looks good 👍 let's attach it to this PR

StorytellerCZ reviewed (approved):
Just need more documentation on migration. Other than that I'm happy to move this forward.

jankapunkt commented on package.js:
```diff
@@ -1,33 +1,36 @@
   npmDependencies: {
-    "node-sass": '9.0.0',
-    "@babel/runtime": "7.24.5"
+    '@babel/runtime': '7.26.0',
```
Is it necessary to pin this version? Isn't it installed on app-level anyway?

illusionfield replied:
You're absolutely right! However, without this, the TinyTest GUI crashes. 

It's only needed for that, and I agree that it's quite wasteful. The entire test setup should be modernized as well.

jankapunkt replied:
You can define depnedbcies as test only by calling npm depends within onUse and onTest separately as long as there is no npm.depends outside of them

illusionfield replied:
Thanks for the info, I checked it out!

We can actually drop it entirely because I couldn't get it to work with Meteor 2.8 anyway. The minimum supported Meteor version is now 2.10, and I tested it without the dependency there—everything worked fine, just like with Meteor 3.

jankapunkt reviewed (approved):
I would definitely move forward with this but I would publish an alpha version first, so we can field test this with some real-world apps.

illusionfield committed:
remove @babel/runtime dependency

https://github.com/Meteor-Community-Packages/meteor-scss/pull/323#discussion_r1944648741

illusionfield commented:
The issue with Meteor 2.8 compatibility was that when I published the package with a Meteor 3.x release, it wasn’t compatible with `fibers` in a 2.8 installation. On the other hand, if I published it with a 2.8 release, it couldn’t find `fibers` in installations above 3.0.

This makes sense given the structural changes in Meteor 3, but I couldn’t find a real solution to make it work seamlessly across both versions. That’s why I set the minimum supported Meteor version to 2.10.

I found more details about this issue here:  
[https://forums.meteor.com/t/solved-is-it-possible-to-publish-a-meteor-compiler-package-that-works-for-both-2-0-and-3-0-yes](https://forums.meteor.com/t/solved-is-it-possible-to-publish-a-meteor-compiler-package-that-works-for-both-2-0-and-3-0-yes)

StorytellerCZ commented:
@jankapunkt now that `@babel/runtime` has been remove, I think we can move to releasing a beta and move this forward to full and proper release. We will ask people to test and give feedback again on Friday and then let's move to final release quickly.

DblK commented:
I was looking to move to the next version to avoid issue with install of python for node-sass, but one key feature is missing `includePaths`.
Do you plan to add it back for final `5.0.0` version?

illusionfield commented:
@DblK

This feature has theoretically lost its purpose in 5.0 since predefined search paths are no longer necessary for relative imports.

The `@import` (or now `@use`) works in multiple ways:
- Relative paths: `../../example`
- Project absolute path: `'{}/imports/...'`
- Node modules: `~pkg` (resolves to `project/node_modules/pkg`)
- Absolute file system paths: `"/home/user/dev/globalstyles"` or `"C:\Styles\fonts"`

However, if there are specific cases where `includePaths` is still needed, let’s discuss it. There might be some edge cases I haven’t considered.

DblK commented:
I do understand there are some workaround and I can manage this, however I have one file called `_mixins.scss` and I have dozen of `scss` refering to it by `@import "mixins"` and my `includePaths` set to `['{}/client']`.
The only way to make it work is to update all the scss files to `@import "{}/client/mixins"` (or better use `@use` for better compatibility for the next version.)

I just wanted to avoid the use of `{}/client/` in every single `scss` file. `includePaths` was very helpful for that. But if there is no other way..

And I am lucky if my `includePaths` was `['{}/some/very/long/path/for/holding/my/mixins/folder']`, it would have been more painful to add it in every single scss file 😅.

illusionfield commented:
I see your point, and I totally understand why `includePaths` was useful in that case.

A good alternative is using `@forward` and `@use`, which are the recommended approaches in modern Sass versions. Instead of relying on `includePaths`, you can create a central file (e.g., `client/mixins/_index.scss`) and forward your mixins like this:

```scss
@forward "{}/some/very/long/path/for/holding/my/mixins/folder";
```

Then, in your SCSS files, instead of @import "mixins", you can simply use:

```scss
@use "{}/client/mixins" as *;
````

This way, all your mixins and variables are available everywhere without having to update each file individually. It’s more modular and keeps your paths clean while ensuring better compatibility with future Sass versions.

I hope I understood your issue correctly, but if not, let’s discuss it further! 😊

Maybe it's worth considering a hot prefix for `"{}/client/"` and `"{}/imports/"`, similar to how `~` works for `"{}/node_modules/"`, as an alternative to `includePaths`.  

@StorytellerCZ, what do you think about this?  

Whichever direction we take, I believe this feature discussion should be moved to a separate issue.

StorytellerCZ commented:
Don't really have opinion here. Since we are doing a major version upgrade it is allowed to break some things, just need to ensure that it is properly documented and mentioned in the migration docs.
If it is not much hassle easing the transition by this is fine with me, but then I would be for that we highlight it as such and that people shouldn't expect this to work in the next major release.

dr-dimitru commented:
@illusionfield I agree with @StorytellerCZ that major bump will require detailed migration guidelines and/or references to "modern" SCSS/SASS practices, especially on including/importing files

illusionfield commented:
@dr-dimitru 
I agree, and I've already started working on a draft version ([here](https://github.com/Meteor-Community-Packages/meteor-scss/pull/323#discussion_r1946693456)). It looks like some additional adjustments will be needed, especially regarding the `includePaths` issue.  

I wouldn’t rule out the possibility of it being merged back soon, but there's also a critical bug that just came up ([here](https://github.com/Meteor-Community-Packages/meteor-scss/issues/324)), so there's still some work to do to ensure everything is in good shape for everyone.

dr-dimitru commented on README.md:
````diff
@@ -25,132 +21,112 @@ Package.onUse(function (api) {
-meteor add seba:minifiers-autoprefixer
+```bash
+export DEBUG_PACKAGE_SASS=true
 ```
````
@illusionfield I'd suggest keeping short description about prefixer in the main Readme file, for example as [in my PR](https://github.com/Meteor-Community-Packages/meteor-scss/pull/325/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R154-R162)

dr-dimitru commented:
@illusionfield I left comments for migration doc and changes in readme

dr-dimitru commented on README.md:
```diff
@@ -25,132 +21,112 @@ Package.onUse(function (api) {
-<thead>
-<tr><th>Meteor Version</th><th>Recommended fourseven:scss version</th></tr>
-</thead>
-<tbody>
-<tr><td>1.0 - 1.1</td><td>3.2.0</td></tr>
-<tr><td>1.2 - 1.3.1</td><td>3.4.2</td></tr>
-<tr><td>1.3.2+</td><td>3.8.0_1</td></tr>
-<tr><td>1.4.0</td><td>3.8.1</td></tr>
-<tr><td>1.4.1+</td><td>4.5.4</td></tr>
-<tr><td>1.6+</td><td>4.12.0</td></tr>
-</tbody>
-</table>
```
@illusionfield I'm suggesting to keep this table for historical reference and to provide support for older projects

illusionfield replied:
@dr-dimitru I have restored it as you suggested.

dr-dimitru reviewed (changes requested):
Please see my comments regarding documentation

illusionfield committed:
Migration Guide

illusionfield committed:
Internal Meteor Packages: improve root file detection logic for imports and package files
https://github.com/Meteor-Community-Packages/meteor-scss/issues/324

illusionfield committed:
Implement includePaths option
https://github.com/Meteor-Community-Packages/meteor-scss/pull/323#issuecomment-2659045373

illusionfield committed:
Merge branch 'master' into develop

illusionfield committed:
Merge branch 'hotfix/internal-packages'

illusionfield committed:
Merge branch 'master' into feature/documentation

* master:
  Internal Meteor Packages: improve root file detection logic for imports and package files https://github.com/Meteor-Community-Packages/meteor-scss/issues/324

illusionfield committed:
readme draft

illusionfield committed:
Merge branch 'feature/documentation' into develop

illusionfield commented:
### Updates in this PR

- The `includePaths` option is now available again. (@DblK)
- Fixed issue #324.
- The **Migration Guide** has been completed.

### Help Needed

@dr-dimitru, @StorytellerCZ

I would appreciate some help with finalizing the **README**.  
I'm unsure about what key points to highlight and how to effectively focus on the features introduced with this update without repeating content already covered in the **Migration Guide**.

Any suggestions or feedback would be greatly appreciated!

dr-dimitru commented:
@illusionfield good, will review shortly. In general it's okay to repeat some key things, where migration guide will be very detailed document

dr-dimitru commented on MIGRATION-GUIDE.md:
I'm suggesting to add version name to Migration-guide file-name

dr-dimitru commented on MIGRATION-GUIDE.md:
```diff
@@ -0,0 +1,193 @@
+
+**Important:** The minimum required Meteor version is 2.10, and the package fully supports Meteor 3.
+
+### 🚨 Critical: Remove Seba Minifiers and Use Standard Minifier CSS 🚨
```
Change title to "Critical: Remove outdated packages"

dr-dimitru commented on MIGRATION-GUIDE.md:
````diff
@@ -0,0 +1,193 @@
+
+```sh
+meteor update fourseven:scss
+```
````
Add removal instructions right to the first section, as most users will quit after "update package"

```sh
meteor remove seba:minifiers-autoprefixer
meteor add standard-minifier-css
```

dr-dimitru commented on MIGRATION-GUIDE.md:
```diff
@@ -0,0 +1,193 @@
+- This transition ensures **better compatibility and long-term maintainability**.
+
+## Migration Steps
+
```
Add ToC between these two headers to indicate this is multi-step procedure:

- Step 1: Update...
- Step 2: remove outdated...
- Step 3: Install and use NPM packages for....
- Step 4: Change `@imports` to `@use`
- Step 5: Update variables usage

Each line can link to headers in this document

dr-dimitru commented on MIGRATION-GUIDE.md:
````diff
@@ -0,0 +1,193 @@
+```sh
+meteor remove seba:minifiers-autoprefixer
+meteor add standard-minifier-css
+meteor npm install --save-dev autoprefixer postcss postcss-load-config
````
Installing and configuring new autoprefixer should have separate section

dr-dimitru commented on MIGRATION-GUIDE.md:
```diff
@@ -0,0 +1,193 @@
+
+### Configuration File Update
+
+The configuration file has been renamed from `scss-config.json` to `.scss.config.json` to support a wider range of Dart Sass-specific settings.
```
Propose or reference recommended config file

dr-dimitru commented on MIGRATION-GUIDE.md:
```diff
@@ -0,0 +1,193 @@
+| Config File Name       | `scss-config.json`                      | `.scss.config.json`                      |
+| Debug Mode             | Not available                           | Enabled via environment variable         |
+
+This guide helps developers transition to the Dart Sass-based `fourseven:scss` package, ensuring compatibility with Meteor 3 and modern Sass features.
```
Add "Further reading" section at the end of the document with:
1. Sum up of all used external links in the document
2. Links to mentioned packages
3. Related links where devs can learn more

dr-dimitru commented on README.md:
```diff
@@ -1,20 +1,20 @@
-**Meteor 1.7 introduced a change in how node_modules are handled, if you want to import sass from a node_module you need to symlink the package in your imports directory (more information below)**
+This is a major change—from LibSass to Dart Sass. Please read the [Migration Guide](MIGRATION-GUIDE.md) for detailed instructions before proceeding.
+
+This is a build plugin for Meteor.js that compiles Sass files using Dart Sass.
```
Let's keep this line right after opening `<h1>` header "Dart Sass for Meteor.js"

dr-dimitru commented on README.md:
```diff
@@ -1,20 +1,20 @@
+## Important Notice
 
-**Meteor 1.7 introduced a change in how node_modules are handled, if you want to import sass from a node_module you need to symlink the package in your imports directory (more information below)**
+This is a major change—from LibSass to Dart Sass. Please read the [Migration Guide](MIGRATION-GUIDE.md) for detailed instructions before proceeding.
```
Let's use `[!IMPORTANT]` quote style of GitHub, I updated text a little as well:

> [!IMPORTANT]  
> In `v5.0.0` this package has introduced a major change — migrating underlying library `LibSass` to `Dart Sass`. Please read the [Migration Guide](MIGRATION-GUIDE.md) for detailed instructions before proceeding

Markup:

```markdown
> [!IMPORTANT]  
> In `v5.0.0` this package has introduced a major change — migrating underlying library `LibSass` to `Dart Sass`. Please read the [Migration Guide](MIGRATION-GUIDE.md) for detailed instructions before proceeding
```

dr-dimitru commented on README.md:
```diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
-imports.
+You can import styles from various locations, including other packages, your own app, and npm modules.
+
+- **Importing from another package**:
```
Better to use 4th level header here `####`

dr-dimitru commented on README.md:
````diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
 ```
 
-Importing styles from the target app:
+- **Importing from your app**:
````
Better to use 4th level header here `####`

dr-dimitru commented on README.md:
````diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
 ```
 
-This can also conveniently be used to import styles from npm modules for example:
+- **Importing from npm modules**:
````
Better to use 4th level header here `####`

dr-dimitru commented on README.md:
```diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
+
+To use Autoprefixer, follow the official [PostCSS with Meteor](https://docs.meteor.com/packages/standard-minifier-css.html#standard-minifier-css) documentation.
+
+## Custom Configuration
+
+You can customize the behavior of the Sass compiler by creating a `.scss.config.json` or `scss-config.json` file in the root of your project. This file supports options like `quietDeps`, `verbose`, and others that control Sass compilation behavior.
+
+Example configuration:
```
Is there reference to config file on Sass/Dart Sass resources that we can link here?

illusionfield replied:
yes, here:
[SASS Compile Options](https://sass-lang.com/documentation/js-api/interfaces/options/)
but currently only these 2 options are implemented. [See here](https://github.com/illusionfield/fourseven-scss/blob/092e7dba0bad192d328bcf7866fc760b35d1f168/plugin/compile-scss.js#L186)

dr-dimitru commented on README.md:
````diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
-```
+Source maps are enabled by default, helping with debugging by mapping the compiled CSS back to the original Sass files.
+
+### Autoprefixer
+
+To use Autoprefixer, follow the official [PostCSS with Meteor](https://docs.meteor.com/packages/standard-minifier-css.html#standard-minifier-css) documentation.
````
I'd add autoprefixer installation here as Meteor's article focused more on PostCSS

```sh
meteor npm install --save-dev autoprefixer postcss postcss-load-config
```

The rest can be learned from PostCSS Meteor docs article

dr-dimitru commented on README.md:
```diff
@@ -25,132 +25,116 @@ Package.onUse(function (api) {
+### Global Include Path
 
-See the [meteor changelog](https://github.com/meteor/meteor/blob/devel/History.md) for more information.
+At the moment, there is no support for a global include path. If there is significant demand, this feature could be added in future updates.
```
Isn't `includePaths` with `@forward` acts as global import?

illusionfield replied:
Currently, it seems necessary to support the `includePaths` option because, without it, the transition would place an even greater burden on users, who already face a significant effort in adapting their projects.  

Additionally, some level of support for the [Sass migrator](https://sass-lang.com/documentation/cli/migrator/) might still be required to facilitate the migration process.

dr-dimitru reviewed (changes requested):
@illusionfield I left comments regarding docs. Please review, — I'm open for discussion

illusionfield commented:
> @illusionfield I left comments regarding docs. Please review, — I'm open for discussion

@dr-dimitru Thank you very much, it's awesome! 

However, the README is in serious need of refactoring, and I'm afraid I might be short on time due to my deadline-driven work commitments. As a result, this PR will likely remain in a pending state, even though it forms the foundation of all my projects. :)

Another observation is that, since the package was almost entirely rewritten from scratch, the tests have become quite superficial. They do not cover many expected and unexpected edge cases, which makes even the smallest modifications require extensive manual testing. Keeping this in mind, there is still a lot of work ahead before the release. 

I appreciate any help in this process!

jankapunkt commented:
Any demand for another package release? Are we still in alpha or is this already beta stage?

illusionfield commented:
@jankapunkt 
Yes, we definitely need another alpha release since `includePaths` has been reintroduced and issue #324 has been fixed.  

I see the need for the community to test it, but unfortunately, the `README.md` is in a pretty bad state.

The code itself is in beta state—every feature has been implemented and tested. We are only dependent on the documentation, which is still in an early stage.

dr-dimitru commented:
@illusionfield I can take on docs. Let me know where I can push my changes; @StorytellerCZ @jankapunkt or perhaps we can merge this PR as it is into separate branch (like `feature/dart-sass/5.0.0`) and continue working on it

illusionfield commented:
Thank you so much, @dr-dimitru! 

This is really a great help. Feel free to reach out if you have any questions about how the code works.  

A `release/5` branch has been created from this PR—should I set the PR target to it?  

It’s also very important to update the package tests, and I’d be happy to put some effort into that.

jankapunkt committed:
published 5.0.0-alpha.2

jankapunkt commented:
released 5.0.0-alpha.2 but accidentally pushed the commit to `dat-css`, please merge that branch, I will delete it once we have merged everything into Master at the very end.

illusionfield committed:
fix #324
fully tested!

illusionfield committed:
version bump

illusionfield committed:
Merge branch 'dat-css'

illusionfield commented:
@jankapunkt  I have successfully merged the dat-css branch into my branch and also bumped the version to 5.0.0-alpha.3.

illusionfield commented:
Issue #326 has been raised about the migration guide not being accessible in the package registry. Since this is an important part of the transition, we should ensure it's correctly linked. What is the best practice for referencing a non-README file within an ongoing PR?

@dr-dimitru @StorytellerCZ @jankapunkt

dr-dimitru commented:
@illusionfield use this link to the guide on your Fork — https://github.com/illusionfield/fourseven-scss/blob/master/MIGRATION-GUIDE.md

dr-dimitru commented:
> Thank you so much, @dr-dimitru!
> 
> This is really a great help. Feel free to reach out if you have any questions about how the code works.
> 
> A `release/5` branch has been created from this PR—should I set the PR target to it?
> 
> It’s also very important to update the package tests, and I’d be happy to put some effort into that.

Any chance to get write permissions to your fork to manage changes to the docs?

illusionfield commented:
@dr-dimitru 
Sorry for the delay. I have just sent the invitation!

dr-dimitru commented:
> @dr-dimitru Sorry for the delay. I have just sent the invitation!

Well received, thanks! I'll try to propose changes this week

illusionfield committed:
windows compatibility fix

https://github.com/Meteor-Community-Packages/meteor-scss/issues/327

illusionfield commented:
I (hopefully) fixed the issue #327 as well. Dart Sass has a significantly different path resolver ([Dart Sass Path Resolver](https://github.com/dart-lang/core/blob/0b2bd3fcd7f3e082f4cc9b14c19ffa93894b85ae/pkgs/path/lib/path.dart#L474)) compared to the Meteor package API's path handling ([Meteor Package API](https://github.com/meteor/meteor/blob/devel/docs/source/api/packagejs.md#accessing-file-system)).

I concluded that it makes sense to merge the two and utilize Dart Sass's path resolver, as it significantly improves the plugin's speed and expands the possibilities of path resolving without needing to implement each option separately.

This is yet another step towards the beta version.

@dr-dimitru If you can clean up the documentation chaos I created, it would be worth releasing the next version.

@jankapunkt Currently, the version is 5.0.0-alpha.3, but once the documentation is ready, it would make sense to release it as 5.0.0-beta.1.

illusionfield commented:
[Here](https://github.com/Meteor-Community-Packages/meteor-scss/issues/327#issuecomment-2686019067)
a feature was mentioned that isn't well documented but seems interesting. It might be worth extending the documentation to cover this as well.

@dr-dimitru , @StorytellerCZ  what do you think?

illusionfield committed:
decode URI components in loaded URLs for correct file path handling

https://github.com/Meteor-Community-Packages/meteor-scss/issues/327

jankapunkt commented:
@illusionfield just let me know when there is a beta release to be published, I will then review and publish.

illusionfield commented:
> @illusionfield just let me know when there is a beta release to be published, I will then review and publish.

@dr-dimitru when do you think the documentation is likely to be ready?
Thanks again for your help! 🙏🙏🙏

illusionfield committed:
include paths `Scss compiler Error: Invalid URL:` bugfix

https://github.com/Meteor-Community-Packages/meteor-scss/issues/324

illusionfield committed:
Update documentation

https://github.com/Meteor-Community-Packages/meteor-scss/pull/323#pullrequestreview-2621354448
https://github.com/Meteor-Community-Packages/meteor-scss/pull/323/files/139f02f0023293b2a045c96d6217e28ab2dcd8bd

illusionfield commented:
I have updated the documentation—it's not necessarily perfect, but should be a good improvement. I have also fixed all issues related to the 5.0 release. The current version is **5.0.0-alpha.3**.

@jankapunkt I believe the package is ready for the beta release.

jankapunkt commented:
I released `5.0.0-alpha.3` as this version was set in `package.js` but there was no release for it. Wen treat it like a beat though :-D

illusionfield commented:
I hope the PR can be merged soon, as I would like to switch back from the temporary alternative to the official version in my projects. 😃

illusionfield committed:
update SASS to v1.85.1

illusionfield committed:
Add GitHub Actions test suite

illusionfield committed:
Bump version to 5.0.0-beta.1

illusionfield committed:
Migration Guide typos correction

illusionfield commented:
I took some inspiration from here: https://github.com/veliovgroup/flow-router/pull/123  
Hopefully, this will only make the package even stronger.

illusionfield commented:
Hi everyone,  

Some time has passed, and the latest release seems to be working fine.  

@dr-dimitru, @StorytellerCZ Do you have any feedback or observations regarding the documentation?  

@jankapunkt If the documentation is also in good shape, then maybe this could be **rc.0** instead of beta.  
What would be the next steps to get this PR merged into the main branch as soon as possible?  

Additionally, I have a question:  
I have an advanced development version that no longer includes Sass as an npm dependency but allows it to be installed separately in `node_modules`. This is important because I use [Dart Sass Embedded](https://www.npmjs.com/package/sass-embedded) in my projects.
How would it be best to integrate this as an official option so it can evolve alongside this package if needed?

jankapunkt commented:
> @jankapunkt If the documentation is also in good shape, then maybe this could be rc.0 instead of beta.

no problem

>  What would be the next steps to get this PR merged into the main branch as soon as possible?

We first need to make sure we have at least one or two more people testing this as part of their running apps. I have two mid-sized apps with bootstrap and some custom SCSS and would like to check this out. However, I'm currently occupied with some other things at work. 
Therefore I kindly ask @dr-dimitru and @StorytellerCZ to do some testing with their apps, if possible.
I just want to avoid this getting merged too soon and then incoming issue reports building immediate pressure (which I will not be able to handle then)

> Additionally, I have a question: I have an advanced development version that no longer includes Sass as an npm dependency but allows it to be installed separately in node_modules. This is important because I use [Dart Sass Embedded](https://www.npmjs.com/package/sass-embedded) in my projects. How would it be best to integrate this as an official option so it can evolve alongside this package if needed?

How is this integrated within your project? Simply by installing the package or by some "linking"? If you need to link this somewhere then I would go for a dependency-injection.

illusionfield commented:
@jankapunkt 
I must confess—I "borrowed" the idea from an official Meteor package 😄  
[standard-minifier-css](https://github.com/meteor/meteor/blob/2d9cd4f1a06163f4c9d40abf9e317d3850fb2b2f/packages/standard-minifier-css/plugin/postcss.js#L29).  

Right now, it's still for [private use](https://github.com/illusionfield/meteor-dart-sass-development), and I'm evaluating whether it makes sense to proceed in this direction.  

I'm already using it in a few projects, and it works—but of course, that doesn't necessarily mean it's suitable for the official community package. That still needs to be determined.

dr-dimitru commented:
> > @jankapunkt If the documentation is also in good shape, then maybe this could be rc.0 instead of beta.
> 
> no problem

I'll review/push changes sometime soon. Wrapping up now with my packages that await for update by community

> > What would be the next steps to get this PR merged into the main branch as soon as possible?
> 
> We first need to make sure we have at least one or two more people testing this as part of their running apps. I have two mid-sized apps with bootstrap and some custom SCSS and would like to check this out. However, I'm currently occupied with some other things at work. Therefore I kindly ask @dr-dimitru and @StorytellerCZ to do some testing with their apps, if possible. I just want to avoid this getting merged too soon and then incoming issue reports building immediate pressure (which I will not be able to handle then)

No problem, will test and report here soon

> > Additionally, I have a question: I have an advanced development version that no longer includes Sass as an npm dependency but allows it to be installed separately in node_modules. This is important because I use [Dart Sass Embedded](https://www.npmjs.com/package/sass-embedded) in my projects. How would it be best to integrate this as an official option so it can evolve alongside this package if needed?
> 
> How is this integrated within your project? Simply by installing the package or by some "linking"? If you need to link this somewhere then I would go for a dependency-injection.

@illusionfield Same question here, if it will be solved with simple `install development npm package X` in the docs — It's okay, if it's more than simply installing a package — I'd link it as dependency

illusionfield commented:
@dr-dimitru 

The integration is as simple as running: 

```sh
meteor npm install –save-dev sass
```

or

```sh
meteor npm install –save-dev sass-embedded
```

No additional linking or special setup is required. 


## There are multiple reasons for this approach

- **Our company does not use a dedicated UI framework** (like Bootstrap, etc.), but instead, we have a custom framework built directly on raw Sass. Because of this, we pay extra attention to how Sass is handled in our workflow.

- **Dart Sass Embedded is significantly faster** than regular Dart Sass, but it is **not platform-independent**. Even `node-gyp` doesn’t solve this because it handles platform dependencies via `optionalDependencies` ([reference here](https://github.com/sass/embedded-host-node/blob/1ec3fbdf241267ce3c55eae4a4fb6d374ed370d9/package.json#L40C4-L40C24)).  
  This means bundling it into the Meteor build **is not feasible** since `publish-for-arch` is no longer available.

- The `quietDeps` option in Sass **only mutes warnings for dependencies** ([see docs](https://sass-lang.com/documentation/js-api/interfaces/options/#quietDeps)).
  - In version **1.79**, it did **not** include `@import` deprecation warnings.
  - However, in **^1.80**, `@import` deprecation warnings were introduced.
  - In our older projects where we haven’t yet migrated `@import` to `@use` or `@forward`, we **temporarily force** version **1.79** to **avoid excessive console warnings** while focusing on refactoring.

illusionfield committed:
ci(test): add support for Meteor 2.10 in GitHub Actions matrix

illusionfield committed:
fix: resolve npm dependency installation issue in Meteor 2 package tests

Addressed an issue where running `meteor test-packages` in Meteor 2 failed due to npm dependency installation errors related to the `meteor-node-stubs` update. This fix ensures compatibility and resolves the 'npmignore: command not found' error during package tests.

For more details, see: https://github.com/meteor/meteor/issues/13691

illusionfield committed:
feat!: decouple Sass version management from plugin (v5.1.0-beta.1)

- Removed `sass` as a direct dependency to allow full control over Sass versioning
- Enables usage of both `sass` and `sass-embedded` via devDependencies
- Updated README.md for improved clarity, examples, and migration guidance
- Bumped version to `5.1.0-beta.1` to reflect breaking internal change

illusionfield committed:
Revert "fix: resolve npm dependency installation issue in Meteor 2 package tests"

This reverts commit 122359fe99473e500b46f6d1b636e7843ef7cef4.

illusionfield committed:
typo

illusionfield committed:
Merge branch 'develop' into feature/remove-sass

illusionfield committed:
Merge branch 'feature/remove-sass' into develop

illusionfield committed:
v5.0.0 Beta 1

illusionfield commented:
Closing this PR in favor of the consolidated version here:  
➡️ **#329 – Finalize Dart Sass migration & drop direct dependency**

That PR targets `release/5.0.0` and includes all changes from this one, along with updated docs and migration instructions.  
Looking forward to your feedback there!
